src/twisted/trial/__main__.py

```diff
@@ -4,6 +4,18 @@
 if __name__ == "__main__":
     import sys
 
-    from pkg_resources import load_entry_point
+    if sys.version_info >= (3, 8):
+        from importlib.metadata import distribution
+    else:
+        from importlib_metadata import distribution
 
-    sys.exit(load_entry_point("Twisted", "console_scripts", "trial")())
+    ep_name = "trial"
+    dist = distribution("twisted")
+    for ep in dist.entry_points:
+        if ep.group == "console_scripts" and ep.name == ep_name:
+            trial = ep
+            break
+    else:
+        raise OSError(f"No console_scripts entry point found for: {ep_name}")
+
+    sys.exit(trial.load()())
```

Comment:
If we're not going to test the error message, my preference would be for something briefer that won't cause issues with coverage, like this:

```suggestion
    sys.exit(
        next(
            ep
            for ep in distribution("twisted").entry_points
            if (ep.group, ep.name) == ("console_scripts", "trial")
        ).load()()
    )
```

(It's not clear to me how we *would* test this message though, unless we factored it out into a function…)